R/inputs-spectrum.R

```diff
@@ -95,6 +95,252 @@ extract_pjnz_one <- function(pjnz) {
+  ## - Else, if number entered for total children 0-14 on ART, use that
+  ## - Else, if percentage entered for children 0-14 on ART, use percentage
+  ##   times children needing ART.
+  ##
+  ## Children needing ART taken from line 9 of <ChildARTCalc MV2>, interpolated
+  ## to end of year. I am not entirely sure what this calculation is, but
+  ## averaging between two years in the internal values matches the values
+  ## reported in the Spectrum ART results table for "Children needing ART (0-14)".
+  ##
+  ## The Spectrum output for number of children on ART in the table can be quite
+  ## different from the internal number on ART in the projection period. I am
+  ## not sure what explains the difference; it should be minimal impact here.
+
+  child_art_raw<- dpsub("<ChildTreatInputs MV3>", 3:6, timedat.idx)
+  child_art_raw <- sapply(child_art_raw, as.numeric)
+  child_art_raw[child_art_raw == -9999] <- NA_real_
+  colnames(child_art_raw) <- proj.years
+
+  child_art_0to14 <- child_art_raw[1,]
+  child_art_aggr <- colSums(child_art_raw[2:4,])
+
+  ## Note: these errors should never occur; don't need to be translated.
+  if (any(is.na(child_art_0to14) & is.na(child_art_aggr))) {
+    stop(paste0("No child ART input found for year: ",
+                paste0(names(which(is.na(child_art_0to14) & is.na(child_art_aggr))), collapse = ", "),
+                ". Check PJNZ file inputs"))
+  }
+
+  if (any(!is.na(child_art_0to14) & !is.na(child_art_aggr))) {
+    stop(paste0("Both aggregated and age-stratified child ART input found for year: ",
+                paste0(names(which(!is.na(child_art_0to14) & !is.na(child_art_aggr))), collapse = ", "),
+                ". Check PJNZ file inputs"))
+  }
+
+  child_art_isperc <- dpsub("<ChildARTByAgeGroupPerNum MV2>", 3, timedat.idx)
+  child_art_isperc <- as.integer(child_art_isperc)
+  names(child_art_isperc) <- proj.years
+
+  child_art_need<- dpsub("<ChildARTCalc MV2>", 9, timedat.idx)
+  child_art_need <- as.numeric(child_art_need)
+  child_art_need <- approx(proj.years, child_art_need, proj.years + 0.5, rule = 2)$y
+  names(child_art_need) <- proj.years
+
+  child_art <- dplyr::case_when(child_art_isperc == 0  & !is.na(child_art_aggr) ~ child_art_aggr,
+                                child_art_isperc == 0  & !is.na(child_art_0to14) ~ child_art_0to14,
+                                child_art_isperc == 1 ~ child_art_need * child_art_0to14 / 100)
+  names(child_art) <- proj.years
+
+  if (any(is.na(child_art))) {
+    stop("Something has gone wrong extracting child ART inputs; please seek troubleshooting.")
+  }
+
+  child_art <- data.frame(sex = "both",
+                          age_group = "Y000_014",
+                          year = proj.years,
+                          art_dec31 = child_art)
+
+  art_dec31 <- rbind(child_art, art15plus)
+
+  art_dec31
+}
+
+#' Disaggregate the number on ART Dec 31 to single age
+#'
+#' @noRd
+#' 
+add_dec31_art <- function(spec, pjnz) {
```

Comment:
Maybe me being silly but I can't see where this function is used?

Reply:
Sorry -- this code got copied over from the v2.6.0 update in #274; it is used there. I think it will create a bit of merge havoc to remove it here now (which will later get merged into that branch). Okay to leave it in here?